### ISSUE TITLE
Parallel test with selenium fails

### ISSUE THREAD

**Issue by m3nthal** [pytest-dev/pytest#1926]
The following code was run with `-n 2 --browser Chrome`:
```
import pytest

@pytest.fixture
def selenium(selenium):
    return selenium

def test_one(selenium):
    selenium.get("http://www.python.org")

def test_two(selenium):
    selenium.get("http://www.google.com")
```
Two browser windows are opened and the the pages were loaded. Then the execution was terminated:

```
============================= test session starts ==============================
platform darwin -- Python 3.5.1, pytest-2.9.0, py-1.4.31, pluggy-0.3.1
driver: Chrome
sensitiveurl: .*
rootdir: /Users/username/python/parallel_test, inifile: 
plugins: allure-adaptor-1.7.5, base-url-1.1.0, html-1.10.0, selenium-1.3.1, variables-1.4, xdist-1.15.0
gw0 I / gw1 I
gw0 [2] / gw1 [2]

scheduling tests via LoadScheduling
INTERNALERROR> Traceback (most recent call last):
INTERNALERROR>   File "/Users/username/.pyenv/versions/3.5.1/lib/python3.5/site-packages/_pytest/main.py", line 94, in wrap_session
INTERNALERROR>     session.exitstatus = doit(config, session) or 0
INTERNALERROR>   File "/Users/username/.pyenv/versions/3.5.1/lib/python3.5/site-packages/_pytest/main.py", line 125, in _main
INTERNALERROR>     config.hook.pytest_runtestloop(session=session)
INTERNALERROR>   File "/Users/username/.pyenv/versions/3.5.1/lib/python3.5/site-packages/_pytest/vendored_packages/pluggy.py", line 724, in __call__
INTERNALERROR>     return self._hookexec(self, self._nonwrappers + self._wrappers, kwargs)
INTERNALERROR>   File "/Users/username/.pyenv/versions/3.5.1/lib/python3.5/site-packages/_pytest/vendored_packages/pluggy.py", line 338, in _hookexec
INTERNALERROR>     return self._inner_hookexec(hook, methods, kwargs)
INTERNALERROR>   File "/Users/username/.pyenv/versions/3.5.1/lib/python3.5/site-packages/_pytest/vendored_packages/pluggy.py", line 333, in <lambda>
INTERNALERROR>     _MultiCall(methods, kwargs, hook.spec_opts).execute()
INTERNALERROR>   File "/Users/username/.pyenv/versions/3.5.1/lib/python3.5/site-packages/_pytest/vendored_packages/pluggy.py", line 596, in execute
INTERNALERROR>     res = hook_impl.function(*args)
INTERNALERROR>   File "/Users/username/.pyenv/versions/3.5.1/lib/python3.5/site-packages/xdist/dsession.py", line 536, in pytest_runtestloop
INTERNALERROR>     self.loop_once()
INTERNALERROR>   File "/Users/username/.pyenv/versions/3.5.1/lib/python3.5/site-packages/xdist/dsession.py", line 555, in loop_once
INTERNALERROR>     call(**kwargs)
INTERNALERROR>   File "/Users/username/.pyenv/versions/3.5.1/lib/python3.5/site-packages/xdist/dsession.py", line 593, in slave_slavefinished
INTERNALERROR>     assert not crashitem, (crashitem, node)
INTERNALERROR> AssertionError: ('test.py::test_one', <SlaveController gw0>)

========================= no tests ran in 2.83 seconds =========================
```

`pip list`:
apipkg (1.4)
beautifulsoup4 (4.5.1)
enum34 (1.1.6)
execnet (1.4.1)
lxml (3.6.4)
namedlist (1.7)
pip (8.1.2)
py (1.4.31)
pyrsistent (0.11.13)
pytest (2.9.0)
pytest-allure-adaptor (1.7.5)
pytest-base-url (1.1.0)
pytest-html (1.10.0)
pytest-selenium (1.3.1)
pytest-variables (1.4)
pytest-xdist (1.15.0)
PyYAML (3.11)
requests (2.10.0)
selenium (2.53.5)
setuptools (18.2)
six (1.10.0)

**Comment by nicoddemus** [comment]
Thanks for the report.

I assume it works if you execute without `-n0`?

**Comment by nicoddemus** [comment]
Btw, that means that the test somehow crashed when executed in parallel, does pytest-selenium support xdist? @davehunt

**Comment by davehunt** [comment]
We're using pytest-selenium with pytest-xdist without any issues, and have been in one form or another for over 4 years. One difference I note is that the reporter is using python 3.5, which we have yet to switch to. Also, we don't use the pytest-allure-adapter, and use a later version of pytest.

**Comment by dshuga** [comment]
I'm seeing something pretty similar in certain circumstances when using xdist and flaky. I'm also using Python 3.5, I haven't yet tried another version. The circumstance is described in a ticket I filed with flaky: https://github.com/box/flaky/issues/117

It's interesting that in my example I'm not using selenium, and in your example you're not using flaky.

**Comment by jakeczyz** [comment]
I saw the same thing (cross-posted https://github.com/box/flaky/issues/117). My env:
`platform linux -- Python 3.6.0, pytest-3.0.7, py-1.4.33, pluggy-0.4.0 metadata: {'Python': '3.6.0', 'Platform': 'Linux-4.4.41-k8s-x86_64-with', 'Packages': {'pytest': '3.0.7', 'py': '1.4.33', 'pluggy': '0.4.0'}, 'Plugins': {'xdist': '1.15.0', 'tap': '2.0', 'metadata': '1.3.0', 'html': '1.14.2', 'flaky': '3.3.0'}}`

**Comment by nicoddemus** [comment]
@davehunt should we move this discussion over pytest-dev/pytest-selenium?

**Comment by davehunt** [comment]
@nicoddemus I've not seen anything that convinces me this is a pytest-selenium issue. The latest comment shows pytest-selenium is not installed, ditto for the comment before it.

**Comment by nicoddemus** [comment]
Oh OK, thanks Dave!

**Comment by nicoddemus** [comment]
@m3nthal can you post a reproducible example? Unfortunately that error is quite generic and means that an internal error has occurred.

If you do find a minimal example, could you post a new issue in https://github.com/pytest-dev/pytest-xdist/issues? Thanks!

**Comment by RonnyPfannschmidt** [comment]
closing due to lack of further input
